packages/java/endpoint/src/test/java/com/vaadin/hilla/crud/CrudRepositoryServiceJpaTest.java

```diff
@@ -93,4 +98,20 @@ public void deleteAll() {
                 testCrudRepositoryService.list(Pageable.unpaged(), null)
                         .stream().map(o -> o.getName()).toList());
     }
+
+    @Test
+    public void listWithMaxPagination() {
+        TestPropertyValues.of("spring.data.rest.pageable.max-page-size=2")
+                .applyTo((ConfigurableEnvironment) applicationContext
+                        .getEnvironment());
+        // force init again, as it already happened
+        testCrudRepositoryService.init();
+
+        List<TestObject> result = testCrudRepositoryService
+                .list(Pageable.ofSize(5), null);
+
+        Assert.assertEquals(2, result.size());
+        Assert.assertEquals("John", result.get(0).getName());
+        Assert.assertEquals("Jeff", result.get(1).getName());
```

Comment:
Isn't there a chance that the repository return the object in a different order and make this test flaky? I suggest removing these two asserts, as asserting the size is enough for the purpose of this testing this functionality.

Reply:
I don't think so, but I agree that they can be removed.